src/topology/algebra/infinite_sum.lean

```diff
@@ -1207,3 +1208,216 @@ lemma dist_le_tsum_dist_of_tendsto₀ [pseudo_metric_space α] {f : ℕ → α}
 by simpa only [zero_add] using dist_le_tsum_dist_of_tendsto h ha 0
 
 end cauchy_seq
+
+/-! ## Multipliying two infinite sums
+
+In this section, we prove various results about `(∑' x : β, f x) * (∑' y : γ, g y)`. Not that we
+always assume that the family `λ x : β × γ, f x.1 * g x.2` is summable, since there is no way to
+deduce this from the summmabilities of `f` and `g` in general, but if you are working in a normed
+space, you may want to use the analogous lemmas in `analysis/normed_space/basic`
+(e.g `tsum_mul_tsum_of_summable_norm`).
+
+We first establish results about arbitrary index types, `β` and `γ`, and then we specialize to
+`β = γ = ℕ` to prove the Cauchy product formula (see `tsum_mul_tsum_nat`).
+
+### Arbitrary index types
+-/
+
+section tsum_mul_tsum
+
+variables [topological_space α] [regular_space α] [semiring α] [topological_semiring α]
+  {f : β → α} {g : γ → α} {s t u : α}
+
+lemma has_sum.mul_eq (hf : has_sum f s) (hg : has_sum g t)
+  (hfg : has_sum (λ (x : β × γ), f x.1 * g x.2) u) :
+  s * t = u :=
+have key₁ : has_sum (λ b, f b * t) (s * t),
+  from hf.mul_right t,
+have this : ∀ b : β, has_sum (λ c : γ, f b * g c) (f b * t),
+  from λ b, hg.mul_left (f b),
+have key₂ : has_sum (λ b, f b * t) u,
+  from has_sum.prod_fiberwise hfg this,
+has_sum.unique key₁ key₂
+
+lemma has_sum.mul (hf : has_sum f s) (hg : has_sum g t)
+  (hfg : summable (λ (x : β × γ), f x.1 * g x.2)) :
+  has_sum (λ (x : β × γ), f x.1 * g x.2) (s * t) :=
+let ⟨u, hu⟩ := hfg in
+(hf.mul_eq hg hu).symm ▸ hu
+
+/-- Product of two infinites sums indexed by arbitrary types.
+    See also `tsum_mul_tsum_of_summable_norm` if `f` and `g` are abolutely summable. -/
+lemma tsum_mul_tsum (hf : summable f) (hg : summable g)
+  (hfg : summable (λ (x : β × γ), f x.1 * g x.2)) :
+  (∑' x, f x) * (∑' y, g y) = (∑' z : β × γ, f z.1 * g z.2) :=
+hf.has_sum.mul_eq hg.has_sum hfg.has_sum
+
+end tsum_mul_tsum
+
+section cauchy_product
+
+/-! ### `ℕ`-indexed families (cauchy product) -/
+
+variables {f : ℕ → α} {g : ℕ → α}
+
+open finset
+
+/-- The equivalence behind the Cauchy product formula. It maps `(n, k, l)` with `k + l = n` to
+    the pair `(k, l)`, which allows us to show (under appropriate summability assumptions) that
+    `(∑' z : ℕ × ℕ, f z.1 * g z.2) = ∑' n : ℕ, ∑ kl in nat.antidiagonal n, f kl.1 * g kl.2`
+    by rearranging terms. -/
+def cauchy_product_equiv : (Σ (n : ℕ), nat.antidiagonal n) ≃ ℕ × ℕ :=
```

Comment:
```suggestion
@[simps] def cauchy_product_equiv : (Σ (n : ℕ), nat.antidiagonal n) ≃ ℕ × ℕ :=
```

Reply:
I think this might make more sense nearer `nat.antidiagonal`, as `sigma_antidiagonal_equiv_prod`, and a comment like "the disjoint union of the antidiagonals is equivalent to the product"